src/qibo/parallel.py

```diff
@@ -177,46 +97,18 @@ def parallel_parametrized_execution(circuit, parameters, initial_state=None, pro
         from qibo.backends import GlobalBackend
         backend = GlobalBackend()
 
-    if not isinstance(parameters, list):  # pragma: no cover
+    if not isinstance(parameters, list):
         from qibo.config import raise_error
         raise_error(RuntimeError, "parameters must be a list.")
 
-    _check_parallel_configuration(processes, backend)
-
-    def operation(params, circuit, state): # pragma: no cover
+    def operation(params, circuit, state):
+        if state is not None:
+            state = backend.cast(state)
```

Comment:
I would think that we need
```suggestion
            state = backend.cast(state, copy=True)
```
here but strangely the test passes as it is now, even if I change the test state to complex type.